charts/gateway/templates/gemfire/embedded/gemfire-tls-secret.yaml

```diff
@@ -13,12 +14,14 @@ metadata:
     {{- end }}
   {{- if  .Values.additionalAnnotations }}
   annotations:
-{{- range $key, $val := .Values.additionalAnnotations }}
+    "helm.sh/hook": pre-install,post-upgrade
+    "helm.sh/hook-weight": "-5"
+  {{- range $key, $val := .Values.additionalAnnotations }}
     {{ $key }}: "{{ $val }}"
-{{- end }}
-{{- end }}
+  {{- end }}
+  {{- end }}
 type: Opaque
-data:
-  truststore: {{ .Values.config.gemfire.external.ssl.truststore | b64enc }}
-  keystore: {{ .Values.config.gemfire.external.ssl.keystore | b64enc }}
+stringData:
+  password: {{ .Values.clusterPassword }}
```

Comment:
This could be auto-generated - should have its own field in values.yaml, should be referencable from an existingSecret

If existingGatewaySecret is used (recommended) this value will not be available or will default to `mypassword`